src/components/svgs/nodes/chongqingrt2021-num-line-badge.tsx

```diff
@@ -0,0 +1,130 @@
+    );
+    const onPointerMove = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerMove(id, e),
+        [id, handlePointerMove]
+    );
+    const onPointerUp = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerUp(id, e),
+        [id, handlePointerUp]
+    );
+
+    const fgColor = color[3];
+    const fontSize = !Number.isInteger(num) ? 15 : 16;
+    const [letterSpacing, sX] = Number.isInteger(num) ? (Number(num) >= 10 ? [-1.2, 1.5] : [0, 5.5]) : [0, 2.55];
+
+    return (
+        <g
+            id={id}
+            transform={`translate(${x - 10.5}, ${y - 10.5})`}
+            onPointerDown={onPointerDown}
+            onPointerMove={onPointerMove}
+            onPointerUp={onPointerUp}
+            style={{ cursor: 'move' }}
+        >
+            <rect fill={color[2]} x="0" width="21" height="21" rx="3" ry="3" />
+            <rect strokeWidth="1.5" stroke="white" fill="none" x="1.5" y="1.5" width="18" height="18" rx="2" ry="2" />
+            <text
+                className="rmp-name__zh"
+                textAnchor="left"
+                x={sX + 0.5}
+                y="10.5"
+                fill={fgColor}
+                fontSize={fontSize}
+                letterSpacing={letterSpacing}
+                dominantBaseline="central"
+            >
+                {num}
+            </text>
+        </g>
+    );
+};
+
+/**
+ * BerlinUBahnLineBadge specific props.
+ */
+export interface ChongqingRT2021NumLineBadgeAttributes extends AttributesWithColor {
+    num: number | string;
+}
+
+const defaultChongqingRT2021NumLineBadgeAttributes: ChongqingRT2021NumLineBadgeAttributes = {
+    num: 1,
+    color: [CityCode.Chongqing, 'cq1', '#e4002b', MonoColour.white],
+};
+
+const chongqingRT2021NumLineBadgeFields = [
+    {
+        type: 'input',
+        label: 'panel.details.nodes.common.num',
+        value: (attrs?: ChongqingRT2021NumLineBadgeAttributes) =>
+            (attrs ?? defaultChongqingRT2021NumLineBadgeAttributes).num,
+        validator: (val: string) => !Number.isNaN(val),
+        onChange: (val: string | number, attrs_: ChongqingRT2021NumLineBadgeAttributes | undefined) => {
+            // set default value if switched from another type
+            const attrs = attrs_ ?? defaultChongqingRT2021NumLineBadgeAttributes;
+            // set value
+            if (Number.isNaN(Number(val))) {
+                attrs.num = val;
+            } else {
+                attrs.num = Number(val);
+            }
+            // return modified attrs
+            return attrs;
+        },
+    },
+    {
+        type: 'custom',
+        label: 'color',
+        component: (
+            <ColorField
+                type={MiscNodeType.ChongqingRT2021NumLineBadge}
+                defaultTheme={defaultChongqingRT2021NumLineBadgeAttributes.color}
+            />
+        ),
+    },
+];
+
+const attrsComponent = () => (
+    <RmgFieldsFieldSpecificAttributes
```

Comment:
`RmgFieldsFieldSpecificAttributes` is deprecated and should not be used anymore :)